cxx-sensors/src/main/java/org/sonar/cxx/sensors/pclint/CxxPCLintSensor.java

```diff
@@ -109,21 +133,67 @@ public void stream(SMHierarchicCursor rootCursor) throws XMLStreamException {
               }
 
               CxxReportIssue issue = new CxxReportIssue(id, file, line, msg);
-              saveUniqueViolation(context, issue);
+
+              if (currentIssue != null) {
+                saveUniqueViolation(context, currentIssue);
+              }
+
+              currentIssue = issue;
             } else {
               LOG.warn("PC-lint warning ignored: {}", msg);
               if (LOG.isDebugEnabled()) {
                 LOG.debug("File: {}, Line: {}, ID: {}, msg: {}", file, line, id, msg);
               }
             }
           }
+
+          if (currentIssue != null) {
+            saveUniqueViolation(context, currentIssue);
+          }
         } catch (com.ctc.wstx.exc.WstxUnexpectedCharException
           | com.ctc.wstx.exc.WstxEOFException
           | com.ctc.wstx.exc.WstxIOException e) {
           LOG.error("Ignore XML error from PC-lint '{}'", CxxUtils.getStackTrace(e));
         }
       }
 
+      private void addMoreLocationsToCurrentIssue(@Nonnull CxxReportIssue currentIssue,
+                                                String file,
+                                                String line,
+                                                String msg) {
+        if (file != null && file.isEmpty() && msg != null && msg.startsWith(PREFIX_DURING_SPECIFIC_WALK_MSG)) {
+          String walkedSrcFile =
+                  msg.substring(PREFIX_MSG_LENGTH, msg.indexOf(" ", PREFIX_MSG_LENGTH));
+
+          Matcher matcher = FILE_NAME_WITH_LINE_AND_COL_PATTERN.matcher(walkedSrcFile);
+
+          if (matcher.matches()) {
+            file = matcher.group(1);
+            line = matcher.group(2);
+          }
+        }
+
+        if (file == null || file.isEmpty() || line == null || line.isEmpty()) {
+          return;
+        }
+
+        assert !currentIssue.getLocations().isEmpty() : "current issue must have a location.";

```

Comment:
@bloodlee you should not assert here. Write a warning or error to the LOG file instead. Please have a look to other sensors how this works. In case of error seems also https://github.com/SonarOpenCommunity/sonar-cxx/wiki/Supported-configuration-properties#sonarcxxerrorrecoveryenabled.

Reply:
Agreed! I will update my implementation.